torch/csrc/jit/passes/symbolic_shape_analysis.h

```diff
@@ -0,0 +1,22 @@
+#pragma once
+
+#include <torch/csrc/WindowsTorchApiMacro.h>
+#include <torch/csrc/jit/ir/ir.h>
+
+namespace torch {
+namespace jit {
+
+// CAUTION NOT TO BE USED, STILL A WIP, NOT STABLE
+
+TORCH_API void PropagateShapesWithShapeFunction(
+    Node* n,
+    const std::shared_ptr<Graph>& graph);
```

Comment:
I know you asked not to comment on the API, but for my better understanding - do we want to keep that as an external API?

Reply:
This is just an API for testing really I can delete this API, i don't think it's used at the moment. Eventually you will register shape functions for particular schemas and then we will have an API for propagating shapes on a graph or node, so users should never be directly passing in a shape function.